src/effects/audiounits/AudioUnitEffect.cpp

```diff
@@ -658,6 +658,17 @@ int AudioUnitEffect::ShowClientInterface(
    return dialog.ShowModal();
 }
 
+EffectSettings AudioUnitEffect::MakeSettings() const
+{
+   auto result = StatefulPerTrackEffect::MakeSettings();
+   // Cause initial population of the map stored in the stateful effect
+   if (!mInitialFetchDone) {
+      FetchSettings(GetSettings(result));
+      mInitialFetchDone = true;
+   }
+   return result;
+}
+
```

Comment:
`mInitialFetchDone` is a member variable, which controls if local variable `result` will be fetched or not. This looks suspicious. First call to `MakeSettings` will give back an actual fetched result, but following ones will not? or am I missing something?

Reply:
There is not real statelessness yet.  There is only one real AudioUnitEffectSettings (for each effect).  One time only for that effect, fetch settings so that the map is populated; but give back a dummy EffectSettings `result` that doesn't really contain the state.

Reply:
Do this side effect before the first `StoreSettings` happens, otherwise `StoreSettings` fails to do anything.

As I recall, your bug reproduced only when pluginsettings.cfg was clean.  But if the graphic EQ had been applied destructively in a previous run of Audacity, then it was not clean.